compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/contracts/FirAbstractContractResolveTransformerDispatcher.kt

```diff
@@ -61,12 +62,25 @@ abstract class FirAbstractContractResolveTransformerDispatcher(
     private val regularDeclarationsTransformer = FirDeclarationsResolveTransformer(this)
 
     private var contractMode = true
+    private var insideContractDescription = false
 
     override fun transformAnnotation(annotation: FirAnnotation, data: ResolutionMode): FirStatement {
         return annotation
     }
 
     override fun transformAnnotationCall(annotationCall: FirAnnotationCall, data: ResolutionMode): FirStatement {
+        // Annotation calls inside contracts are not allowed and at this point we are also
+        // unable to resolve them. We indicate this by returning a resolved error type.
+        if (insideContractDescription) {
+            val typeDiagnostic = ConeSimpleDiagnostic(
+                "Cannot infer annotation call type during CONTRACTS phase.",
+                DiagnosticKind.InferenceError
```

Comment:
Could you please introduce a new `DiagnosticKind` here, like `DiagnosticKind.AnnotationInContract`?
`InferenceError` is used for errors in type inference, and there is no inference here